common/util/iterator_range.h

```diff
@@ -51,14 +51,19 @@ class iterator_range {
   iterator end_;
 };
 
+template <typename Iter>
+iterator_range(Iter, Iter) -> iterator_range<Iter>;
```

Comment:
Can we double check whether this is supported in C++17 (if that is our current supported standard)?

Reply:
C++17 final draft: https://timsong-cpp.github.io/cppwp/n4659/temp.deduct.guide
Yes, Verible requires C++17.

Reply:
We also compile with `--cxxopt=-Wctad-maybe-unsupported` to check for implicit deduction without guide, so since this deduction guide is added here, that should work, and the CI will tell.

(What we should do, in a separate cleanup PR, is to use `iterator_range()` everywhere and remove `make_range()` to minimize confusion about what to use)